From 1baee8d8f28d0e5f047f390877216abf74f31592 Mon Sep 17 00:00:00 2001
From: Fedora systemd team <systemd-maint@redhat.com>
Date: Tue, 24 Nov 2015 09:20:31 +0100
Subject: [PATCH] Revert "core: one step back again, for nspawn we actually
 can't wait for cgroups running empty since systemd will get exactly zero
 notifications about it"

This reverts commit 743970d2ea6d08aa7c7bff8220f6b7702f2b1db7.
---
 src/core/unit.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/core/unit.c b/src/core/unit.c
index b9e1f13..85294fa 100644
--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -3545,16 +3545,7 @@ int unit_kill_context(
                         if (r != -EAGAIN && r != -ESRCH && r != -ENOENT)
                                 log_unit_warning_errno(u->id, r, "Failed to kill control group: %m");
                 } else if (r > 0) {
-
-                        /* FIXME: For now, we will not wait for the
-                         * cgroup members to die, simply because
-                         * cgroup notification is unreliable. It
-                         * doesn't work at all in containers, and
-                         * outside of containers it can be confused
-                         * easily by leaving directories in the
-                         * cgroup. */
-
-                        /* wait_for_exit = true; */
+                        wait_for_exit = true;
 
                         if (c->send_sighup && k != KILL_KILL) {
                                 set_free(pid_set);
-- 
2.5.0

